src/cluster/cluster_manager.rs

```diff
@@ -158,9 +154,11 @@ impl ClusterManager {
                     update = cluster_updates_rx.recv() => {
                         match update {
                             Some(update) => {
-                                Self::update_cluster_update_metrics(&metrics, &update);
-                                let update = Self::create_endpoints_from_update(&update);
                                 debug!(log, "Received a cluster update.");
+                                let num_clusters = update.len() as i64;
+                                let update = Self::create_endpoints_from_update(update);
+                                let num_endpoints = update.as_ref().map(|ep| ep.as_ref().len() as i64).unwrap_or_default();
+                                Self::update_cluster_update_metrics(&metrics, num_clusters, num_endpoints);
                                 cluster_manager.write().update(update);
```

Comment:
Can we move this to its own `handle_cluster_update` function?

Reply:
Done!